app/client-v2/src/views/HomeView.vue

```diff
@@ -0,0 +1,150 @@
+};
+const apiUrl = getApiUrl();
+const {
+  data,
+  error: projectsError,
+  isFetching,
+  execute: refetchProjects
+} = useFetch(apiUrl + "/projects", { credentials: "include" }).json<ProjectsResponse>();
+
+const filters = ref({
+  name: { value: null, matchMode: FilterMatchMode.CONTAINS }
+});
+
+const newProjectName = ref("");
+const addProject = async () => {
+  if (!newProjectName.value) return;
+  const { error, data: project } = await useFetch(apiUrl + "/project", {
+    credentials: "include"
+  })
+    .post({ name: newProjectName.value })
+    .json();
+
+  if (error.value) {
+    showErrorToast("Error creating project");
+    return;
+  }
+  router.push(`/project/${project.value?.data}`);
+};
+
+const editingRows = ref([]);
+const onRowEditSave = async (event: DataTableRowEditSaveEvent) => {
+  let { newData, index } = event;
+
+  if (data.value !== null && data.value.data[index].name !== newData.name) {
+    const { error } = await useFetch(apiUrl + `/project/${data?.value?.data[index].id}/rename`, {
+      credentials: "include"
+    })
+      .post({ name: newData.name })
+      .json();
+    if (error.value) {
+      showErrorToast("Error renaming project");
+      return;
+    }
+
+    showSuccessToast("Project renamed successfully");
+    refetchProjects();
+  }
+};
+</script>
+
+<template>
+  <Toast />
+  <div class="flex flex-column gap-1 mb-3">
+    <span class="text-3xl font-bold">Projects</span>
+    <span class="text-color-secondary">View and navigate to existing projects or create new ones</span>
+  </div>
+  <div class="surface-card p-4 shadow-2 border-round">
+    <DataTable
+      :value="data?.data"
+      v-model:filters="filters"
+      :pt="{
+        table: { style: 'min-width: 50rem' },
+        header: { style: 'padding: 0.75rem 0rem' },
+        column: {
+          bodycell: ({ state }: Record<string, any>) => ({
+            style: state['d_editing'] && 'padding-top: 0.45rem; padding-bottom: 0.45rem;'
+          })
+        }
+      }"
+      paginator
+      removableSort
+      v-model:editingRows="editingRows"
+      editMode="row"
+      @row-edit-save="onRowEditSave"
+      :loading="isFetching"
+      :rows="10"
+    >
+      <template #header>
+        <div class="flex flex-wrap align-items-center justify-content-between gap-2">
+          <span class="p-input-icon-left">
+            <i class="pi pi-search" />
+            <InputText v-model="filters.name.value" placeholder="Search project by name" />
+            <Button
+              type="button"
+              icon="pi pi-filter-slash"
+              label="Clear"
+              outlined
+              class="ml-2"
+              v-if="filters.name.value"
+              @click="filters.name.value = null"
+            />
+          </span>
+          <div>
+            <InputGroup>
+              <InputText placeholder="Create new Project" v-model="newProjectName" @keydown.enter="addProject" />
```

Comment:
just some consistent casing
```suggestion
              <InputText placeholder="Create new project" v-model="newProjectName" @keydown.enter="addProject" />
```